### ISSUE TITLE
[Rotarycraft v31b] Item Filter GUI Crashes/Doesn't update

### ISSUE THREAD

**Issue by Falgar** [ReikaKalseki/Reika_Mods_Issues#3196]
There are two issues I've encountered with the Item Filter. The first is that when an item is placed in the template slot, the GUI remains blank, and the various filter options only appear if you close and then re-open the GUI. This behavior directly lead me to encounter the second, more serious issue. If, on a blank GUI, either due to having no template item or not re-opening it after inserting a template item, the page next button is pressed twice (I believe setting it to what would be the NBT data page) the game crashes with a NPE, updating screen events.

Here is a pastebin of the relevant bits of the crash log: [https://pastebin.com/mHKUgE1n](https://pastebin.com/mHKUgE1n)

**Comment by ReikaKalseki** [comment]
Fixed.

**Comment by Falgar** [comment]
I just tested this in v32b and while the first issue has been fixed (the updating of the gui when an item is placed in the template slot), the second issue, crashing on pressing the right arrow twice on a blank gui is still present.

**Comment by ReikaKalseki** [comment]
Now those buttons are not present on blank pages.

**Comment by Falgar** [comment]
That is not the case. I just booted into my testworld (running v32b) and the buttons were there and I crashed. Link to the crash report: https://pastebin.com/Lu0sJPYk

There is an additional note however that was previously unable to occur. If an item is placed in the template slot and then removed, pressing the right button twice does not crash, but rather shows 3 tiny icons, (checkmark, x, and question mark), nor does any other page crash it. Upon closing and then reopening the gui, it returns to crashing.
